### PR TITLE
Uses new data platform lib in orc8r-certifier

### PR DESCRIPTION
**Pull request by sanchezfdezjavier**
# Description

Uses new data platform lib in orc8r-certifier.

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

File: nms-magmalte-operator/tests/integration/test_integration.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ async def _deploy_orc8r_certifier(self, ops_test):
             series="jammy",
         )
         await ops_test.model.add_relation(
-            relation1=CERTIFIER_APPLICATION_NAME, relation2="postgresql-k8s:db"
+            relation1=CERTIFIER_APPLICATION_NAME, relation2="postgresql-k8s:postgresql_client"
```
Comment (Gmerold) [inline nms-magmalte-operator/tests/integration/test_integration.py]: Relation name didn't change, so there's no need to update this.

Comment (ghislainbourgeois) [reply nms-magmalte-operator/tests/integration/test_integration.py]: The relation endpoint on `postgresql-k8s` side needs to change however. But not to `postgresql_client`, which is the interface, but to `database`: https://github.com/canonical/postgresql-k8s-operator/blob/b01f121625808ee57eff84cf274896ce8ed4f5cf/metadata.yaml#L32

We will need to update all the PRs to also update the bundle thinking about it.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: nms-magmalte-operator/tests/integration/test_integration.py (hunk 1 above)
Comment (Gmerold) [reply nms-magmalte-operator/tests/integration/test_integration.py]: Yes, for some reason I thought it applies to the certifier side. Stupid... :)

Comment (sanchezfdezjavier) [reply nms-magmalte-operator/tests/integration/test_integration.py]: First, you're right. Second, this file is not meant to be changed in this PR, this is `nms-magmalte`. Reverted.

Comment (Gmerold) [reply nms-magmalte-operator/tests/integration/test_integration.py]: As mentioned by @ghislainbourgeois, we need to update the bundle as well.

**Review by Gmerold — APPROVED** [review]

File: nms-magmalte-operator/tests/integration/test_integration.py (hunk 1 above)
Comment (sanchezfdezjavier) [reply nms-magmalte-operator/tests/integration/test_integration.py]: Done.

File: orc8r-certifier-operator/src/charm.py [hunk 2]
```diff
@@ -128,11 +130,13 @@ def __init__(self, *args):
...
-        )
-        self.framework.observe(
-            self._db.on.database_relation_broken, self._on_database_relation_broken
+            self._database.on.database_created,
+            self._on_database_relation_joined,  # TODO: This should be a different event?
         )
+        # TODO: Handle relation broken event when the new data platform lib supports it.
+        # self.framework.observe(
+        #     self._db.on.database_relation_broken, self._on_database_relation_broken
+        # )
```
Comment (ghislainbourgeois) [inline orc8r-certifier-operator/src/charm.py]: Can't we just hand the relation broken ourselves?

File: orc8r-certifier-operator/src/charm.py [hunk 3]
```diff
@@ -128,11 +130,13 @@ def __init__(self, *args):
 
         # Relation events
         self.framework.observe(
-            self._db.on.database_relation_joined, self._on_database_relation_joined
-        )
-        self.framework.observe(
-            self._db.on.database_relation_broken, self._on_database_relation_broken
+            self._database.on.database_created,
+            self._on_database_relation_joined,  # TODO: This should be a different event?
```
Comment (ghislainbourgeois) [inline orc8r-certifier-operator/src/charm.py]: I am not sure what the `TODO` is meant to say here, are you talking about the method that we call to respond to the event? In this case, I think it should be renamed right away.

File: orc8r-certifier-operator/tests/unit/test_charm.py [hunk 4]
```diff
@@ -658,23 +644,28 @@ def test_given_certificates_relation_not_created_when_on_pebble_ready_then_statu
...
         self.harness.container_pebble_ready(container_name="magma-orc8r-certifier")
 
         assert self.harness.charm.unit.status == BlockedStatus(
             "Waiting for certificates relation to be created"
         )
 
+    # TODO: Handle relation broken event when supported.
+    @pytest.mark.xfail(
+        reason="Currently the new postgres lib does not support relation broken events https://github.com/canonical/data-platform-libs/pull/11/files"  # noqa: E501
+    )
```
Comment (ghislainbourgeois) [inline orc8r-certifier-operator/tests/unit/test_charm.py]: The PR linked here is closed. I think we should handle `relation-broken` directly here.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: orc8r-certifier-operator/metadata.yaml [hunk 5]
```diff
@@ -40,8 +39,8 @@ peers:
     interface: orc8r-certifier-replica
 
 requires:
-  db:
-    interface: pgsql
+  database:
```
Comment (gruyaume) [inline orc8r-certifier-operator/metadata.yaml:42]: We didn't need to change "our" side of the interface

Comment (sanchezfdezjavier) [reply orc8r-certifier-operator/metadata.yaml:42]: I've changed it for the sake of consistency, it's pretty confusing looking at the [postgres-k8s requires](https://github.com/canonical/postgresql-k8s-operator/blob/b01f121625808ee57eff84cf274896ce8ed4f5cf/metadata.yaml#L32) and seeing we match the name for the deprecated relation but the interface for the new one. Let me know if that makes sense.

File: orc8r-certifier-operator/src/charm.py (hunk 2 above)
Comment (sanchezfdezjavier) [reply orc8r-certifier-operator/src/charm.py]: Done.

File: orc8r-certifier-operator/src/charm.py (hunk 3 above)
Comment (sanchezfdezjavier) [reply orc8r-certifier-operator/src/charm.py]: Fixed.

File: orc8r-certifier-operator/tests/unit/test_charm.py (hunk 4 above)
Comment (sanchezfdezjavier) [reply orc8r-certifier-operator/tests/unit/test_charm.py]: Fixed.

File: orc8r-certifier-operator/metadata.yaml (hunk 5 above)
Comment (gruyaume) [reply orc8r-certifier-operator/metadata.yaml:42]: Humm in this case you may want to align with @ghislainbourgeois , [this PR](https://github.com/canonical/charmed-magma-orchestrator/pull/251) for nms-magmalte uses `db` as the relation name.

Comment (sanchezfdezjavier) [reply orc8r-certifier-operator/metadata.yaml:42]: We've already did, we agreed to set them all to `database`, this way we follow the new naming.

**Review by gruyaume — APPROVED** [review]

**Review by ghislainbourgeois — APPROVED** [review]

**Review by ghislainbourgeois — APPROVED** [review]

**Review by Gmerold — APPROVED** [review]
Good job and also good luck with the conflicts with [this guy here](https://github.com/canonical/charmed-magma-orchestrator/pull/253) ;)

**Review by saltiyazan — APPROVED** [review]

**Pull request merged**